### PR TITLE
Nc fix/column delete bug

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

## Summary by CodeRabbit

- **New Features**
	- Introduced input sanitization for URL fields to enhance data validation and formatting.

- **Improvements**
	- Improved data reloading behavior in the toolbar's column filter by specifying data offset.
	- Streamlined handling of various view columns in the database, focusing on efficient data insertion and cache management.

- **Refactor**
	- Simplified code by removing unnecessary cache setting operations across different view models.
	- Adjusted data insertion and caching logic for a more consistent approach in handling metadata.

### CODE REVIEW

**Review by mertmit — APPROVED** [review]

File: packages/nc-gui/components/cell/Url.vue [hunk 1]
```diff
@@ -47,28 +47,30 @@ const isExpandedFormOpen = inject(IsExpandedFormOpenInj, ref(false))!
 
 const isForm = inject(IsFormInj)!
 
+const trimVal = (val:string) => val && (val + '').trim();
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/Url.vue:50]: if the type is `string`, then we don't need to check `val` and add `' '` (?)

Comment (pranavxc) [reply packages/nc-gui/components/cell/Url.vue:50]: it's still possible to have non-string value here so as an extra precaution I kept it.

File: packages/nocodb/src/models/Column.ts [hunk 2]
```diff
@@ -824,61 +824,81 @@ export default class Column<T = any> implements ColumnType {
...
     await ncMeta.metaDelete(null, null, MetaTable.GALLERY_VIEW_COLUMNS, {
-      fk_column_id: col.id,
+      fk_column_id: id,
     });
-    const galleryViewColumnId = await NocoCache.get(
-      `${CacheScope.GALLERY_VIEW_COLUMN}:${col.id}`,
-      CacheGetType.TYPE_STRING,
-    );
-    if (galleryViewColumnId) {
+
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/models/Column.ts:898]: Cache scope value and table name could be different

File: packages/nocodb/src/models/MapViewColumn.ts [hunk 3]
```diff
@@ -57,15 +57,13 @@ export default class MapViewColumn {
       insertObj.source_id = viewRef.source_id;
     }
 
-    const { id, fk_column_id } = await ncMeta.metaInsert2(
+    const { id } = await ncMeta.metaInsert2(
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/models/MapViewColumn.ts:60]: It's not used anywhere in the code block

File: packages/nocodb/src/services/views.service.ts [hunk 4]
```diff
@@ -76,6 +76,10 @@ export class ViewsService {
   }) {
     const model = await Model.get(param.tableId);
 
+    if (!model) {
+      NcError.tableNotFound(param.tableId);
+    }
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/services/views.service.ts:81]: within the static method it already throws the error

**Pull request merged**